### PR TITLE
Use birpc

### PR DESCRIPTION
**Pull request by codetheweb**
Greatly simplifies code and paves the way for us to grab a new database instance within `beforeTemplateIsBaked()` in application code.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -63,71 +114,34 @@ export const getTestPostgresDatabaseFactory = <
...
-          }
+    return {
+      ...connectionDetailsFromWorker,
+      pool,
+      pgbouncerPool,
+    }
+  }
 
+  let rpcCallback: (data: any) => void
+  const rpc = createBirpc<SharedWorkerFunctions, TestWorkerFunctions>(
```
Comment (ethanplee14) [inline src/index.ts:139]: is the idea meant to establish a bidirectional communication with the postgres docker container?

File: src/index.ts [hunk 2]
```diff
@@ -143,94 +157,74 @@ export const getTestPostgresDatabaseFactory = <
...
-          } finally {
-            // Otherwise connection will be killed by worker when converting to template
-            await connectionDetails.pool.end()
-          }
-        }
+          const hookResult = await options.beforeTemplateIsBaked({
+            params: params as any,
+            connection: connectionDetails,
+            containerExec: async (command): Promise<ExecResult> =>
+              rpc.execCommandInContainer(command),
```
Comment (ethanplee14) [inline src/index.ts:164]: you weren't kidding this is definitely way simpler

**Review by ethanplee14 — APPROVED** [review]

File: src/index.ts (hunk 1 above)
Comment (codetheweb) [reply src/index.ts:139]: not quite

AVA's architecture has one worker thread per test file (called a test worker) and one worker thread per "shared worker" (called... a shared worker)

previous to this PR, we implemented 2-way RPC calls between the test worker and shared workers ourselves, but using birpc really makes the intent and logic cleaner

Comment (ethanplee14) [reply src/index.ts:139]: oh i see, this is to communicate with ava's worker thread

File: src/index.ts [hunk 3]
```diff
@@ -143,94 +157,74 @@ export const getTestPostgresDatabaseFactory = <
...
-        return {
-          ...mapWorkerConnectionDetailsToConnectionDetails(
-            replyData.connectionDetails
-          ),
-          beforeTemplateIsBakedResult:
-            replyData.beforeTemplateIsBakedResult.result,
-        }
+    for await (const msg of worker.subscribe()) {
+      if (msg.data.type === "teardown") {
+        console.log("breaking")
```
Comment (phpnode) [inline src/index.ts:198]:
```suggestion
```

Comment (codetheweb) [reply src/index.ts]: whoops thanks

3e31180

**Review by panzerchris — APPROVED** [review]

**Comment by panzerchris** [comment]
Took me a while but this looks like a solid update!

**Pull request merged**